packages/core/src/redocly/cloud/api-client.ts

```diff
@@ -0,0 +1,199 @@
+          Authorization: `Bearer ${this.apiKey}`,
+        },
+        body: JSON.stringify({
+          mountPath: remote.mountPath,
+          mountBranchName: remote.mountBranchName,
+          type: 'CICD',
+          autoMerge: true,
+        }),
+      }
+    );
+
+    try {
+      return await this.getParsedResponse<UpsertRemoteResponse>(response);
+    } catch (err) {
+      throw new Error(`Failed to upsert remote: ${err.message || 'Unknown error'}`);
+    }
+  }
+
+  async push(
+    organizationId: string,
+    projectId: string,
+    payload: PushPayload,
+    files: { path: string; stream: ReadStream | Buffer }[]
+  ): Promise<PushResponse> {
+    const formData = new FormData();
+
+    formData.append('remoteId', payload.remoteId);
+    formData.append('commit[message]', payload.commit.message);
+    formData.append('commit[author][name]', payload.commit.author.name);
+    formData.append('commit[author][email]', payload.commit.author.email);
+    formData.append('commit[branchName]', payload.commit.branchName);
+    payload.commit.url && formData.append('commit[url]', payload.commit.url);
+    payload.commit.namespace && formData.append('commit[namespaceId]', payload.commit.namespace);
+    payload.commit.sha && formData.append('commit[sha]', payload.commit.sha);
+    payload.commit.repository && formData.append('commit[repositoryId]', payload.commit.repository);
+    payload.commit.createdAt && formData.append('commit[createdAt]', payload.commit.createdAt);
+
+    for (const file of files) {
+      formData.append(`files[${file.path}]`, file.stream);
+    }
+
+    payload.isMainBranch && formData.append('isMainBranch', 'true');
+
+    const response = await fetch(
+      `${this.domain}/api/orgs/${organizationId}/projects/${projectId}/pushes`,
+      {
+        method: 'POST',
+        headers: {
+          Authorization: `Bearer ${this.apiKey}`,
+        },
+        body: formData,
+      }
+    );
+
+    try {
+      return await this.getParsedResponse<PushResponse>(response);
+    } catch (err) {
+      throw new Error(`Failed to push: ${err.message || 'Unknown error'}`);
+    }
+  }
+
+  async getRemotesList(organizationId: string, projectId: string, mountPath: string) {
+    const response = await fetch(
+      `${this.domain}/api/orgs/${organizationId}/projects/${projectId}/remotes?filter=mountPath:/${mountPath}/`,
+      {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: `Bearer ${this.apiKey}`,
+        },
+      }
+    );
+
+    try {
+      return await this.getParsedResponse<ListRemotesResponse>(response);
+    } catch (err) {
+      throw new Error(`Failed to get remote list: ${err.message || 'Unknown error'}`);
+    }
+  }
+
+  async getPush({
+    organizationId,
+    projectId,
+    pushId,
+  }: {
+    organizationId: string;
+    projectId: string;
+    pushId: string;
+  }) {
+    const response = await fetchWithTimeout(
```

Comment:
Please take into consideration the possibility of the `push` command working behind a proxy. Please consult to the [original issue](https://github.com/Redocly/redocly-cli/issues/529) and the suggested [fork](https://github.com/joaonlopes/redocly-cli-behind-proxy). 
You might also find this closed [MR](https://github.com/Redocly/redocly-cli/pull/1173) useful in that context.

Reply:
I added `abort-controller` package and moved new client to `cli` package. Also created issue to move old [client and handle proxy](https://github.com/Redocly/redocly-cli/issues/1404)